File: BB.Core/Model/UserReaction.cs
using System;

namespace BB.Core.Model
{
    public class UserReaction
    {
        public Guid UserReactionId { get; set; }

        public long UserID { get; set; }

        public Guid? CommentId { get; set; }

        public Guid? RatingId { get; set; }

        public Reaction Reaction { get; set; }

        public virtual User User { get; set; }

        public virtual Comment Comment { get; set; }

        public virtual Rating Rating { get; set; }
    }
}

File: BB.Web/app/Model/Users/UserReaction.ts
import u = require("./User");
import User = u.User;
import r = require("../Products/Rating");
import Rating = r.Rating;
import rc = require("../Enum/Reaction");
import Reaction = rc.Reaction;
import c = require("../Comment/Comment");
import Comment = c.Comment;

export class UserReaction
{
    public UserReactionId: string;
    public UserID: number;
    public CommentId?: string;
    public RatingId?: string;
    public Reaction: Reaction;
    public User: User;
    public Comment: Comment;
    public Rating: Rating;
}